samples/keys/identity_key_usage/src/main.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <string.h>
+#include <stdio.h>
+#include <psa/crypto.h>
+#include <nrf_cc3xx_platform.h>
+#include <nrf_cc3xx_platform_ctr_drbg.h>
+#include <nrf_cc3xx_platform_defines.h>
+#include <identity_key.h>
+
+#define IDENTITY_KEY_PUBLIC_KEY_SIZE (65)
+uint8_t m_pub_key[IDENTITY_KEY_PUBLIC_KEY_SIZE];
+
+LOG_MODULE_REGISTER(identity_key_usage, LOG_LEVEL_DBG);
+
+void main(void)
+{
+	int err;
+	psa_status_t status;
+	psa_key_attributes_t key_attributes = PSA_KEY_ATTRIBUTES_INIT;
+	psa_key_id_t key_id_out = 0;
+	uint8_t key[IDENTITY_KEY_SIZE_BYTES];
+	size_t olen;
+
+	err = nrf_cc3xx_platform_init();
+	if (err != NRF_CC3XX_PLATFORM_SUCCESS) {
+		LOG_INF("nrf_cc3xx_platform_init returned error: %d", err);
+		return;
+	}
+
+	LOG_INF("Initializing PSA crypto.");
+
+	status = psa_crypto_init();
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_crypto_init returned error: %d", status);
+		return;
+	}
+
+	/* Configure the key attributes for Curve type secp256r1*/
+	psa_set_key_usage_flags(&key_attributes,
+				PSA_KEY_USAGE_SIGN_HASH | PSA_KEY_USAGE_VERIFY_HASH);
+	psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_VOLATILE);
+	psa_set_key_algorithm(&key_attributes, PSA_ALG_ECDSA(PSA_ALG_SHA_256));
+	psa_set_key_type(&key_attributes, PSA_KEY_TYPE_ECC_KEY_PAIR(PSA_ECC_FAMILY_SECP_R1));
+	psa_set_key_bits(&key_attributes, 256);
+
+	if (!identity_key_is_written()) {
+		LOG_INF("No identity key found in KMU (required). Exiting!");
+		return;
+	}
+
+	LOG_INF("Reading the identity key.");
+	err = identity_key_read(key);
+	if (err != ERR_IDENTITY_KEY_SUCCESS) {
+		LOG_INF("Identity key read failed! (Error: %d). Exiting!", err);
+		return;
+	}
+
+	LOG_INF("Importing the identity key into PSA crypto.");
+	status = psa_import_key(&key_attributes, key, IDENTITY_KEY_SIZE_BYTES, &key_id_out);
+	if (status != PSA_SUCCESS) {
+		LOG_INF("psa_import_key failed! (Error: %d). Exiting!", status);
+		return;
+	}
+
+	/* Clear out the key once it is imported */
```

Comment:
Comments should explain why. Not what.

I can clearly see that we are clearing out the key. But why?

Reply:
This is very common in the type of applications. We don't want keys residing in memory we have no guarantees that it is not cleared out. This is a very sensitive key in the system. We can amend the comment with a little more explanation

Will fix!